### PR TITLE
Add Liveness Probe for NEG controller

### PR DESCRIPTION
**Pull request by freehan**

### CODE REVIEW

**Comment by freehan** [comment]
rebased

File: pkg/neg/controller.go [hunk 1]
```diff
@@ -191,7 +211,11 @@ func (c *Controller) serviceWorker() {
 
 // processService takes a service and determines whether it needs NEGs or not.
 func (c *Controller) processService(key string) error {
-	defer lastSyncTimestamp.WithLabelValues().Set(float64(time.Now().UTC().UnixNano()))
+	defer func() {
+		now := c.syncTracker.Track()
+		lastSyncTimestamp.WithLabelValues().Set(float64(now.UTC().UnixNano()))
```
Comment (nicksardo) [inline pkg/neg/controller.go]: What if no services or endpoints. It could happen.

File: pkg/neg/controller.go [hunk 2]
```diff
@@ -154,6 +160,16 @@ func (c *Controller) Run(stopCh <-chan struct{}) {
 	<-stopCh
 }
 
+func (c *Controller) IsHealthy() error {
+	// check if last seen service and endpoint processing is more than an hour ago
+	if c.syncTracker.Get().Before(time.Now().Add(-time.Hour)) {
+		msg := fmt.Sprintf("NEG controller has not proccessed any service and endpoint updates for more than an hour. Something went wrong. Last sync was on %v", c.syncTracker.Get())
```
Comment (nicksardo) [inline pkg/neg/controller.go]: `proccessed`

**Pull request merged**